### DISCUSSION TITLE
HPC computing speed

### DISCUSSION THREAD

**Discussion by Dewu-Yang** [loganoz/horses3d#168] [Q&A] [answered]
Dear all,
I'm trying to perform computations on a cluster, and I'm currently encountering an issue where the efficiency of parallel computing with multiple cores is not high. When I use 600 cores for computation, the speed is very slow. What is the reason？

Best,
Dewu

**Comment by loganoz** [comment]
Hi @Dewu-Yang,

Can you give more details. What speedup are you getting with 600 cores? Are you using MPI / openmp or a combination? If you are using a combination, how many MPI threads and how many openmp threads.

BR,
Gonzalo

**Reply by Dewu-Yang** [reply]
Dear Prof.Gonzalo,

 I am using MPI and performing computations with 600 CPUs, but the speed is slower compared to serial computing (e.g., mpiexec -np 600 ./horses3d *.hsol < ./horses3d *.hsol).

Best,
Dewu

**Reply by loganoz** [reply]
Hi @Dewu-Yang, 

If you are using that command "mpiexec -np 600 ./horses3d *.hsol < ./horses3d *.hsol" you are probably running in only one core. I would suggest you study the documentation of the cluster you are using to run the simulations.
Please, keep me posted.

Best,
Gonzalo

**Reply by Dewu-Yang** [reply]
Dear Prof.Gonzalo,

   I apologize for the misunderstanding caused by my previous response. I indeed am running on a cluster, and I submit a script for execution rather than directly using "mpiexec -np 600 ./horses3d *.hsol". I use a cluster parallel script with a uniform format for execution.This script follows a standardized parallel computing format on the cluster.

Best,
Dewu

**Reply by Dewu-Yang** [reply]
I have encountered a similar issue on my desktop where parallel computing was slower than single-core computing. I would like to know if there are additional steps required for parallel computing. I have set COMM=PARALLEL during the compilation of HORSES3D, and during the run of test cases, it also shows that the computation is being distributed for multi-threaded computing. The software installation and configuration should be correct.

**Reply by Dewu-Yang** [reply, marked as answer]
Dear Prof.Gonzalo,

    The issue has been resolved. It turns out that the configuration of HORSES3D was not successful initially. After properly configuring it, the cluster speed has improved. Thanks!

Best,
Dewu

**Reply by loganoz** [reply]
Good to know. I'll close the discussion then.
